compiler/llstream.nim

```diff
@@ -12,9 +12,9 @@
 import
   pathutils
 
-# support '-d:useGnuReadline' for backwards compatibility:
-when not defined(windows) and (defined(useGnuReadline) or defined(useLinenoise)):
-  import rdstdin
+template imp(x) = import x
+const hasRstdin = compiles(imp(rdstdin))
```

Comment:
Why so hacky?

Reply:
it avoids duplicating whatever logic is in rstdin to tell whether readLineFromStdin is available, eg linenoise being available or not; it is future proof

Reply:
This makes nightlies fail to build.

Reply:
Because there isn't any logic like you mentioned in rdstdin..